### PR TITLE
actually lock atomically on {also,}toStderr

### LINKED ISSUE #2786: log: data race in multi-cpu
https://circleci.com/gh/cockroachdb/cockroach/8304

https://circle-artifacts.com/gh/cockroachdb/cockroach/8304/artifacts/0/tmp/circle-artifacts.bmXCP9T/testrace.log

looks like we're logging something during test setup. logging singleton strikes again!

### PR DESCRIPTION
**Pull request by tbg**
now it should actually fix #2786

### CODE REVIEW

**Comment by bdarnell** [comment]
Also change the comment in `clog.go` to indicate that `{alsoT,t}oStderr` are protected by `mu` (it looks like all existing uses of these variables hold the lock already)

LGTM

File: util/log/logflags/logflags.go [hunk 1]
```diff
@@ -24,7 +24,7 @@ import (
 )
 
 type atomicBool struct {
-	sync.Mutex
+	*sync.Mutex
```
Comment (tamird) [inline util/log/logflags/logflags.go]: nit: sync.Locker

Comment (tschottdorf) [inline util/log/logflags/logflags.go]: done.

**Comment by tamird** [comment]
LGTM

**Pull request merged**